polkadot/node/core/bitfield-signing/src/lib.rs

```diff
@@ -69,35 +75,101 @@ pub enum Error {
 	MpscSend(#[from] mpsc::SendError),
 
 	#[error(transparent)]
-	Runtime(#[from] RuntimeApiError),
+	Runtime(#[from] util::runtime::Error),
 
 	#[error("Keystore failed: {0:?}")]
 	Keystore(KeystoreError),
+
+	#[error("Cannot find block number for given relay parent")]
+	BlockNumberNotFound,
+
+	#[error("Oneshot for receiving response from Chain API got cancelled")]
+	ChainApiSenderDropped(#[source] oneshot::Canceled),
```

Comment:
this error should be fatal, but we could reuse `Oneshot`, other errors probably not. we are not using fatality here yet, so don't know the implications of adding this an error variant here off-hand

Reply:
we are currently simply ignoring errors that come from the tasks that the subsystem spawned (we just drop the result).

I can work on adding fatality here but it would require quite a refactor of the subsystem, since it just uses spawned tasks but is not getting any result from them in the main loop. The way I'd fix this is to add a oneshot channel for each task and to poll them in the main `run` loop.

Honestly it sounds like a bit of an overkill to me, do you have a better idea? Unfortunately, orchestra does not return a `JoinHandle` for us to wait on.

Either way, I'll reuse the `Oneshot` error here and maybe we can move this discussion here: https://github.com/paritytech/polkadot-sdk/issues/823, as I'd like to keep this change out of this PR if possible (it's getting quite large)

Reply:
Yeah, let's tackle this as part of #823. I just wasn't sure about implications of mixing fatal and non-fatal errors here, but since there are dropped, it should be safe at least.